controllers/registerController.js

```diff
@@ -0,0 +1,34 @@
+import { RegisterRepository } from "../repositories/registerRepository.js";
+import { AuthService } from "../services/authService.js";
+import { registerSchema } from "../helpers/schemas.js"
+
+export class RegisterController {
+    constructor() {
+        this.registerRepository = new RegisterRepository();
```

Comment:
Well, usually it comes down to the usersRepository. Typically, you don't want to store your tokens or any other details separated from the users